File: src/weaponinventorymod/gui/WimGuiTooltip.java
package weaponinventorymod.gui;

import com.fs.starfarer.api.ui.TooltipMakerAPI;

final class WimGuiTooltip implements TooltipMakerAPI.TooltipCreator {
    private static final float WIDTH = 320f;

    private final String text;

    WimGuiTooltip(String text) {
        this.text = text == null ? "" : text;
    }

    @Override
    public boolean isTooltipExpandable(Object tooltipParam) {
        return false;
    }

    @Override
    public float getTooltipWidth(Object tooltipParam) {
        return WIDTH;
    }

    @Override
    public void createTooltip(TooltipMakerAPI tooltip, boolean expanded, Object tooltipParam) {
        tooltip.setParaFontDefault();
        tooltip.setParaFontColor(StockReviewStyle.TEXT);
        tooltip.addPara(text, 0f);
    }

    static boolean hasText(String tooltip) {
        return tooltip != null && tooltip.trim().length() > 0;
    }
}

File: src/main/kotlin/weaponsprocurement/gui/WimGuiTooltip.kt
package weaponsprocurement.gui

import com.fs.starfarer.api.ui.TooltipMakerAPI

class WimGuiTooltip(text: String?) : TooltipMakerAPI.TooltipCreator {
    private val text: String = text ?: ""

    override fun isTooltipExpandable(tooltipParam: Any?): Boolean = false

    override fun getTooltipWidth(tooltipParam: Any?): Float = WIDTH

    override fun createTooltip(tooltip: TooltipMakerAPI, expanded: Boolean, tooltipParam: Any?) {
        tooltip.setParaFontDefault()
        tooltip.setParaFontColor(StockReviewStyle.TEXT)
        tooltip.addPara(text, 0f)
    }

    companion object {
        private const val WIDTH = 320f

        @JvmStatic
        fun hasText(tooltip: String?): Boolean = tooltip != null && tooltip.trim().isNotEmpty()
    }
}
